tests/refactor/test_dispatch.py

```diff
@@ -137,27 +139,72 @@ def _main(res_queue, index, dispatch_callback=create_callback, term_event=event,
     for i in range(3):
         assert check_dict[i]["counter"] >= 2, f"Check of counter for dispatcher '{variable}' failed with count = {check_dict[i]['counter']}"
 
-def test_error_adding_item_after_termination() -> None:
+def test_queued_dispatch_error_adding_item_after_termination() -> None:
     trigger = Event()
 
-    dispatcher = Dispatcher(lambda *_: None, ["q"], trigger, False, 5, 2)
+    dispatcher = QueuedDispatcher(lambda *_: None, ["q"], trigger, False, 5, 2)
     dispatcher.start()
 
     trigger.set()
 
     with pytest.raises(RuntimeError):
         dispatcher.add_item("blah", "q", False)
 
-def test_error_attempting_to_use_non_existent_queue() -> None:
+def test_queued_dispatch_error_attempting_to_use_non_existent_queue() -> None:
     trigger = Event()
-    dispatcher = Dispatcher(lambda *_: None, ["q"], trigger, False, 5, 2)
+    dispatcher = QueuedDispatcher(lambda *_: None, ["q"], trigger, False, 5, 2)
     dispatcher.start()
 
     with pytest.raises(KeyError):
         dispatcher.add_item("blah", "z", False)
 
     trigger.set()
 
-    
+
+@pytest.mark.dispatch
+@pytest.mark.parametrize(
+    "append_during_dispatch", (True, False),
+    ids=("pre_append", "append")
+)
+@pytest.mark.parametrize("multiple", (True, False), ids=("multiple", "single"))
+def test_direct_dispatcher(multiple: bool, append_during_dispatch: bool) -> None:
+    n_elements: int = 10
+    time_threshold: float = 1
+
+    start_time = time.time()
+
+    check_dict = {}
+
+    variables = ["lemons"]
+
+    if multiple:
+        variables.append("limes")
+
+    event = Event()
+    dispatchers: list[DirectDispatcher] = []
+
+    for variable in variables:
+        check_dict[variable] = {"counter": 0}
+        def callback(___: list[typing.Any], _: str, args=check_dict, var=variable) -> None:
+            args[var]["counter"] += 1
+        dispatchers.append(
+            DirectDispatcher(callback, [variable], event)
+        )
+
+    if not append_during_dispatch:
+        for i in range(n_elements):
+            for variable, dispatcher in zip(variables, dispatchers):  
+                dispatcher.add_item({string.ascii_uppercase[i % 26]: i}, variable)
+
+    if append_during_dispatch:
+        for i in range(n_elements):
+            for variable, dispatcher in zip(variables, dispatchers):  
+                dispatcher.add_item({string.ascii_uppercase[i % 26]: i}, variable)
```

Comment:
This seems to be doing the same thing regardles of whether `append_during_dispatch` is `True` or `False`.

Reply:
Yep this was a classic copy paste issue